analyzers/src/SonarAnalyzer.CSharp/Rules/PrivateFieldUsedAsLocalVariable.cs

```diff
@@ -191,33 +195,40 @@ bool IsPrecededWithWrite(SyntaxNode readStatementOrArrowExpression)
 
             public override void VisitIdentifierName(IdentifierNameSyntax node)
             {
-                var memberReference = GetTopmostSyntaxWithTheSameSymbol(node);
-                if (memberReference.Symbol == null)
+                if (privateFields.Keys.Any(x => x.Name == node.Identifier.ValueText))
                 {
-                    return;
+                    var memberReference = GetTopmostSyntaxWithTheSameSymbol(node);
+                    if (memberReference.Symbol is IFieldSymbol fieldSymbol
+                        && privateFields.ContainsKey(fieldSymbol))
+                    {
+                        ClassifyFieldReference(semanticModel.GetEnclosingSymbol(memberReference.Node.SpanStart), memberReference);
+                    }
                 }
+                base.VisitIdentifierName(node);
+            }
 
-                var enclosingSymbol = semanticModel.GetEnclosingSymbol(memberReference.Node.SpanStart);
-
-                if (memberReference.Symbol is IFieldSymbol fieldSymbol
-                    && privateFields.ContainsKey(fieldSymbol))
-                {
-                    ClassifyFieldReference(enclosingSymbol, memberReference);
-                }
-                else if (memberReference.Symbol is IMethodSymbol)
+            public override void VisitInvocationExpression(InvocationExpressionSyntax node)
+            {
+                if (node.GetMethodCallIdentifier() is { } methodCallIdentifier
+                    && methodNames.Contains(methodCallIdentifier.ValueText))
                 {
-                    var pseudoStatement = GetParentPseudoStatement(memberReference);
-                    if (pseudoStatement != null)
+                    var memberReference = GetTopmostSyntaxWithTheSameSymbol(node);
+                    if (memberReference.Symbol is IMethodSymbol)
                     {
-                        var invocationsInPseudoStatement = invocations.GetOrAdd(pseudoStatement, x => new HashSet<ISymbol>());
-                        invocationsInPseudoStatement.Add(memberReference.Symbol);
+                        var pseudoStatement = GetParentPseudoStatement(memberReference);
+                        if (pseudoStatement != null)
+                        {
+                            var invocationsInPseudoStatement = invocations.GetOrAdd(pseudoStatement, x => new HashSet<ISymbol>());
+                            invocationsInPseudoStatement.Add(memberReference.Symbol);
```

Comment:
```suggestion
                    invocations.GetOrAdd(pseudoStatement, _ => new HashSet<ISymbol>())
                        .Add(memberReference.Symbol);
```

less code which better describes the intention